Set dpi, start/end sample in plot movie

cdcapano opened the pull request:
This gives adds ability to set the dpi and start/end samples on the command-line for `plot_movie`. Right now the code will always use the last iteration, which is a problem if trying to run plot movie on an result file that hasn't completed yet (since everything after the checkpoint is just 0). I also changed the argument names from "thin-start" and "thin-end" to "start-sample" and "end-sample" to make it more clear what their purpose is.

cdcapano commented:
In the last push, I also added the ability to parallelize the frame production over multiple nodes using python's multiprocessing.

cdcapano commented on bin/inference/pycbc_inference_plot_movie:
```diff
@@ -73,19 +74,22 @@ logging.info('Loading parameters')
 elif opts.frame_number:
     nframes = opts.frame_number
-    thinint = (fp.niterations / (nframes - 1)) - 1
+    thinint = (opts.end_sample - opts.start_sample)/nframes
```
I tried several different things and can't get an error to arise. It looks like even if you specify an end index that's larger than the largest index in the hdf array, it'll just take all samples to the end.

micamu replied:
Have you also tried without specifying start_sample and end_sample? A command line in the same way it would be run with the old version?

cdcapano commented on bin/inference/pycbc_inference_plot_movie:
```diff
@@ -134,7 +138,7 @@ for frame in range(nframes):
     else:
         z = None
-    sample_num = str(frame * thinint + 1)
+    sample_num = str(frame * thinint + opts.start_sample)
```
No, I think it's better to keep the number of samples be the same thing as the iteration number. Otherwise it can get confusing.

cdcapano commented:
@micamu I addressed your comments; I think this is ready to commit now.

cdcapano commented:
@micamu I added the ability to do log-stepping. I also changed it back so that the samples do start from 1 (e.g., iteration 0 corresponds to sample 1, iteration 9 corresponds to sample 10), as you had it originally. I've tested with frame-number specified, frame-step specified, start/end-sample specified and not specified (so that it defaults to the first and last sample), and tested with log-step. All seem to work fine.
